test/uk/gov/hmrc/tai/connectors/JrsConnectorSpec.scala

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.tai.connectors
+
+import com.codahale.metrics.Timer
+import com.github.tomakehurst.wiremock.client.WireMock.{aResponse, get}
+import org.mockito.Mockito.{times, verify, when, reset => resetMock}
+import org.scalatest.concurrent.{IntegrationPatience, ScalaFutures}
+import play.api.Application
+import play.api.inject.guice.GuiceApplicationBuilder
+import play.api.libs.json.{JsValue, Json}
+import uk.gov.hmrc.play.bootstrap.http.HttpClient
+import uk.gov.hmrc.tai.metrics.Metrics
+import uk.gov.hmrc.tai.model.enums.APITypes
+import uk.gov.hmrc.tai.model.{Employers, JrsClaims, YearAndMonth}
+import utils.{BaseSpec, WireMockHelper}
+
+class JrsConnectorSpec extends BaseSpec with WireMockHelper with ScalaFutures with IntegrationPatience {
+
+  override lazy val app: Application = GuiceApplicationBuilder()
+    .configure("microservice.services.coronavirus-jrs-published-employees.port" -> server.port)
+    .build()
+
+  lazy val mockMetrics = mock[Metrics]
+  lazy val httpClient = inject[HttpClient]
+
+  lazy val jrsConnector = new JrsConnector(httpClient, mockMetrics, servicesConfig)
+
+  val jrsClaimsUrl: String =
+    s"/coronavirus-jrs-published-employees/employee/$nino"
```

Comment:
Since we know this API returns 200, 204 401, 403, 500, we should ensure we cover all these cases

Reply:
Done